lib/pure/uri.nim

```diff
@@ -59,6 +59,9 @@ type
     opaque*: bool
     isIpv6: bool # not expose it for compatibility.
 
+  UriParseError* = object of ValueError
```

Comment:
no strong opinion, but isn't ValueError good enough instead of deep exception hierarchies?

Reply:
But I need reraise this exception, So I want to choose an unique exception. I don't want to catch the wrong ValueError caused by `decodeQuery`. Backwards compatibility should be kept.